client/client-lib/src/mint/mod.rs

```diff
@@ -155,6 +153,110 @@ impl MintClient {
         self.context.db.begin_transaction(ModuleRegistry::default())
     }
 
+    /// Adds the final amounts of `change` to the tx before submitting it
+    /// Allows for multiple `change` outputs
+    pub async fn finalize_change(&self, tx: &mut Transaction, change: Vec<Amount>) {
+        let mut change_outputs: Vec<(usize, NoteIssuanceRequests)> = vec![];
+
+        for amount in change {
+            let (issuances, nonces) = self.create_output_coins(amount).await;
+            let out_idx = tx.outputs.len();
+            tx.outputs.push(Output::Mint(MintOutput(nonces)));
+            change_outputs.push((out_idx, issuances));
+        }
+        let txid = tx.tx_hash();
+
+        let mut dbtx = self.context.db.begin_transaction(ModuleRegistry::default());
+
+        // remove the spent ecash from the DB
+        let mut input_ecash: Vec<(Amount, SpendableNote)> = vec![];
+        for input in &tx.inputs {
+            if let Input::Mint(MintInput(notes)) = input {
+                for (amount, note) in notes.clone() {
+                    let key = CoinKey {
+                        amount,
+                        nonce: note.0,
+                    };
+                    let spendable = dbtx
+                        .get_value(&key)
+                        .await
+                        .expect("DB Error")
+                        .expect("Missing note");
+                    input_ecash.push((amount, spendable));
+                    dbtx.remove_entry(&key).await.expect("DB Error");
+                }
+            }
+        }
```

Comment:
While this special case works here I think modules should just mark inputs as spend when they are generated. If that's done inside a DB tx and constructing the FM tx fails it will automatically be rolled back later.